library/core/src/option.rs

```diff
@@ -559,6 +559,34 @@ impl<T> Option<T> {
         }
     }
 
+    /////////////////////////////////////////////////////////////////////////
+    // Setting a new value
+    /////////////////////////////////////////////////////////////////////////
+
+    /// Inserts `v` into the option then returns a mutable reference
+    /// to the contained value.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// #![feature(option_insert)]
+    ///
+    /// let mut o = None;
+    /// let v = o.insert(3);
+    /// assert_eq!(*v, 3);
+    /// ```
+    #[inline]
+    #[unstable(feature = "option_insert", reason = "new API", issue = "none")]
+    pub fn insert(&mut self, v: T) -> &mut T {
+        *self = Some(v);
```

Comment:
nit: can you call this `val` instead? That helps people with intellisense know what the argument is doing, and also shows up in the rustdoc generated docs.
```suggestion
    pub fn insert(&mut self, val: T) -> &mut T {
        *self = Some(val);
```

Reply:
I'll do this, thanks.
I don't commit your suggestion directly because I've already made a few changes.